lib/compat/wordpress-6.5/fonts/fonts.php

```diff
@@ -84,6 +84,35 @@ function gutenberg_create_initial_post_types() {
 	);
 }
 
+/**
+ * Filters the user capabilities to grant the 'upload_fonts' capability as necessary.
+ *
+ * To grant the 'upload_fonts' capability, file modifications must be allowed, the fonts directory must be
+ * writable, and the user must have the 'edit_theme_options' capability.
+ *
+ * @since 6.5.0
+ *
+ * @param bool[] $allcaps An array of all the user's capabilities.
+ * @return bool[] Filtered array of the user's capabilities.
+ */
+function gutenberg_maybe_grant_upload_font_cap( $allcaps, $caps ) {
```

Comment:
When we bring this logic into WP core, we should definitely add some tests for it.

Reply:
I've added the unit tests, @swissspidy.